src/ADS124S08.cpp

```diff
@@ -64,49 +68,156 @@ void ADS124S08::releaseChipSelect(void){
  */
 ADS124S08::ADS124S08(void)
 {
-	pinMode( CS_PIN, OUTPUT );
-	pinMode( START_PIN, OUTPUT );
-	pinMode( RESET_PIN, OUTPUT );
-	pinMode( CKEN_PIN, OUTPUT );
-	pinMode( DRDY_PIN, INPUT );
+	pinMode(CS_PIN, OUTPUT);
+	pinMode(START_PIN, OUTPUT);
+	pinMode(RESET_PIN, OUTPUT);
+	pinMode(CKEN_PIN, OUTPUT);
+	pinMode(DRDY_PIN, INPUT);
+	pinMode(CAN_LED1, OUTPUT);
 
-	digitalWrite( START_PIN, LOW );
-	digitalWrite( RESET_PIN, HIGH );
-	digitalWrite( CKEN_PIN, LOW );
+	digitalWrite(START_PIN, LOW);
+	digitalWrite(RESET_PIN, HIGH);
+	digitalWrite(CKEN_PIN, LOW);
 
 	/* Default register settings */
-	registers[ID_ADDR_MASK] 			= 0x08;
-	registers[STATUS_ADDR_MASK] 	= 0x80;
-	registers[INPMUX_ADDR_MASK]		= 0x01;
-	registers[PGA_ADDR_MASK] 			= 0x00;
+	registers[ID_ADDR_MASK] = 0x08;
+	registers[STATUS_ADDR_MASK] = 0x80;
+	registers[INPMUX_ADDR_MASK] = 0x01;
+	registers[PGA_ADDR_MASK] = 0x00;
 	registers[DATARATE_ADDR_MASK] = 0x14;
-	registers[REF_ADDR_MASK] 			= 0x10;
-	registers[IDACMAG_ADDR_MASK] 	= 0x00;
-	registers[IDACMUX_ADDR_MASK] 	= 0xFF;
-	registers[VBIAS_ADDR_MASK] 		= 0x00;
-	registers[SYS_ADDR_MASK]			= 0x10;
-	registers[OFCAL0_ADDR_MASK] 	= 0x00;
-	registers[OFCAL1_ADDR_MASK] 	= 0x00;
-	registers[OFCAL2_ADDR_MASK] 	= 0x00;
-	registers[FSCAL0_ADDR_MASK] 	= 0x00;
-	registers[FSCAL1_ADDR_MASK] 	= 0x00;
-	registers[FSCAL2_ADDR_MASK] 	= 0x40;
-	registers[GPIODAT_ADDR_MASK] 	= 0x00;
-	registers[GPIOCON_ADDR_MASK]	= 0x00;
+	registers[REF_ADDR_MASK] = 0x10;
+	registers[IDACMAG_ADDR_MASK] = 0x00;
+	registers[IDACMUX_ADDR_MASK] = 0xFF;
+	registers[VBIAS_ADDR_MASK] = 0x00;
+	registers[SYS_ADDR_MASK] = 0x10;
+	registers[OFCAL0_ADDR_MASK] = 0x00;
+	registers[OFCAL1_ADDR_MASK] = 0x00;
+	registers[OFCAL2_ADDR_MASK] = 0x00;
+	registers[FSCAL0_ADDR_MASK] = 0x00;
+	registers[FSCAL1_ADDR_MASK] = 0x00;
+	registers[FSCAL2_ADDR_MASK] = 0x40;
+	registers[GPIODAT_ADDR_MASK] = 0x00;
+	registers[GPIOCON_ADDR_MASK] = 0x00;
 	fStart = false;
 	releaseChipSelect();
 	deassertStart();
-}
-
-void ADS124S08::begin()
-{
-	SPI.begin( CS_PIN );
-	SPI.setBitOrder(MSBFIRST);
-	SPI.setDataMode( SPI_MODE1 );
-	SPI.setClockSpeed( 1000000 );
-	#if defined (SPI_HAS_TRANSACTION)
-		mySPISettings = SPISettings(1000000, MSBFIRST, SPI_MODE1);
-	#endif
+}
+
+void ADS124S08::SPI_Init(uint32_t SPI_speed, uint8_t chipSelect_Pin)

```

Comment:
The library follows standard Arduino conventions using ::begin rather than the very specific SPI_Init